core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -255,9 +255,35 @@ private[spark] object HighlyCompressedMapStatus {
     // we expect that there will be far fewer of them, so we will perform fewer bitmap insertions.
     val emptyBlocks = new RoaringBitmap()
     val totalNumBlocks = uncompressedSizes.length
-    val threshold = Option(SparkEnv.get)
-      .map(_.conf.get(config.SHUFFLE_ACCURATE_BLOCK_THRESHOLD))
-      .getOrElse(config.SHUFFLE_ACCURATE_BLOCK_THRESHOLD.defaultValue.get)
+    val accurateBlockSkewedFactor = Option(SparkEnv.get)
+      .map(_.conf.get(config.SHUFFLE_ACCURATE_BLOCK_SKEWED_FACTOR))
+      .getOrElse(config.SHUFFLE_ACCURATE_BLOCK_SKEWED_FACTOR.defaultValue.get)
+    val shuffleAccurateBlockThreshold =
+      Option(SparkEnv.get)
+        .map(_.conf.get(config.SHUFFLE_ACCURATE_BLOCK_THRESHOLD))
+        .getOrElse(config.SHUFFLE_ACCURATE_BLOCK_THRESHOLD.defaultValue.get)
+    val threshold =
+      if (accurateBlockSkewedFactor > 0) {
+        val sortedSizes = uncompressedSizes.sorted
+        val medianSize: Long = Utils.median(sortedSizes)
```

Comment:
it seems we sort `uncompressedSizes` twice.  The `sortedSizes` is sorted but the `Utils.median` sort it again.

Reply:
Yes, `uncompressedSizes` is sorted twice. Maybe we can change `Utils.median(sizes: Array[Long])` to `Utils.median(sizes: Array[Long], alreadySorted: Boolean = false)`,  and change `Utils.median(sortedSizes)` to `Utils.median(sortedSizes, true)` to avoid this extra sort ? 

@ulysses-you WDYH

Reply:
@wankunde looks good , thanks